src/GafferOSL/OSLShader.cpp

```diff
@@ -256,6 +256,35 @@ Plug *loadStringParameter( const OSLQuery::Parameter *parameter, const InternedS
 	return plug.get();
 }
 
+Plug *loadStringArrayParameter( const OSLQuery::Parameter *parameter, const InternedString &name, Gaffer::Plug *parent )
+{
+	vector<string> defaultValue;
+	if( parameter->sdefault.size() )
+	{
+		defaultValue.resize( parameter->sdefault.size() );
+		for( size_t i = 0; i < parameter->sdefault.size(); i++ )
+		{
+			defaultValue[i] = parameter->sdefault[i].c_str();
+		}
+	}
+
+	StringVectorDataPtr defaultValueData = new StringVectorData( defaultValue );
```

Comment:
Can we allocate the StringVectorData first, and then fill `writable()`, rather than need to copy construct the contents here?